frontend/src/app/util/shape_icon.cljs

```diff
@@ -0,0 +1,83 @@
+;; This Source Code Form is subject to the terms of the Mozilla Public
+;; License, v. 2.0. If a copy of the MPL was not distributed with this
+;; file, You can obtain one at http://mozilla.org/MPL/2.0/.
+;;
+;; Copyright (c) KALEIDOS INC
+
+(ns app.util.shape-icon
+  (:require
+   [app.common.types.component :as ctk]
+   [app.common.types.shape :as cts]
+   [app.common.types.shape.layout :as ctl]))
+
+(defn get-shape-icon*
+  "Returns the icon for a shape based on its type and properties"
+  [shape main-instance?]
```

Comment:
The `main-instance?` can already be obtained from shape, so having it in an additional parameter looks strange